src/runtime/composables/useUI.ts

```diff
@@ -1,24 +1,29 @@
 import { computed, toValue, useAttrs } from 'vue'
+import type { Ref } from 'vue'
 import { useAppConfig } from '#imports'
 import { mergeConfig, omit, get } from '../utils'
 import { Strategy } from '../types'
 
-export const useUI = <T>(key, $ui: Partial<T & { strategy: Strategy }>, $config?: T, { mergeWrapper = false }: { mergeWrapper?: boolean } = {}) => {
+export const useUI = <T>(key, $ui: Ref<Partial<T & { strategy: Strategy }> | undefined>, $config?: T, $wrapperClass?: Ref<string>) => {
   const $attrs = useAttrs()
   const appConfig = useAppConfig()
 
-  const ui = computed(() => mergeConfig<T>(
-    $ui?.strategy || (appConfig.ui?.strategy as Strategy),
-    mergeWrapper ? { wrapper: $attrs?.class } : {},
-    $ui || {},
-    process.dev ? get(appConfig.ui, key, {}) : {},
-    toValue($config || {})
-  ))
+  const ui = computed(() => {
+    const _ui = toValue($ui)
+    const _wrapperClass = toValue($wrapperClass)
+    return mergeConfig<T>(
+      _ui?.strategy || (appConfig.ui?.strategy as Strategy),
+      _wrapperClass ? { wrapper: _wrapperClass } : {},
+      _ui || {},
+      process.dev ? get(appConfig.ui, key, {}) : {},
+      $config || {}
+    )
+  })
+
   const attrs = computed(() => omit($attrs, ['class']))
```

Comment:
Do we still need this as `class` won't be present in `$attrs` since defined in props?

Reply:
I think it can prevent unexpected class merging when one forgets to add `class` to props.